trace/config.go

```diff
@@ -203,3 +205,56 @@ func (i instrumentationVersionOption) ApplyTracer(config *TracerConfig) {
 }
 
 func (instrumentationVersionOption) private() {}
+
+// ErrorConfig is a group of options for error.
+type ErrorConfig struct {
+	Code    codes.Code
+	Message string
+	// EventOpts will be passed to addEvent
+	EventOpts []EventOption
+}
+
+// ErrorOption applies an option to a ErrorConfig.
+type ErrorOption interface {
+	ApplyError(config *ErrorConfig)
+
+	// A private method to prevent users implementing the
+	// interface and so future additions to it will not
+	// violate compatibility.
+	private()
+}
+
+// NewErrorConfig applies all the options to a returned ErrorConfig.
+func NewErrorConfig(options ...ErrorOption) *ErrorConfig {
+	c := new(ErrorConfig)
+	for _, option := range options {
+		option.ApplyError(c)
+	}
+	return c
+}
+
+type statusErrorOption struct {
+	Code    codes.Code
+	Message string
+}
+
+func (o statusErrorOption) ApplyError(c *ErrorConfig) {
+	c.Code = o.Code
+	c.Message = o.Message
+}
+func (statusErrorOption) private() {}
+
+// WithErrorStatus set the error code and message when code is not codes.Unset
+func WithErrorStatus(code codes.Code, message string) ErrorOption {
+	return statusErrorOption{code, message}
+}
+
+type eventOptsErrorOption []EventOption
+
+func (o eventOptsErrorOption) ApplyError(c *ErrorConfig) { c.EventOpts = o }
+func (eventOptsErrorOption) private()                    {}
+
+// WithEventOpts set event options, will be passed to addEvent
+func WithEventOpts(option ...EventOption) ErrorOption {
```

Comment:
```suggestion
func WithEventOptions(option ...EventOption) ErrorOption {
```

If we can't get rid of this I'd like it to not use an abbreviated name.

Reply:
Thanks very much.